Hydra-node crashes with segmentation fault

Hello,

I am trying to run a Hydra node (version 0.7.4) on an Ubuntu 18.04 workstation with go version 1.17.7. 

For some reason I am getting a segmentation fault immediately upon launching the node (please see the error below). I would appreciate any help or insight you might provide. 

Thanks!  

$ go run main.go -ui-theme=none
🐉 Hydra Booster starting up...
📊 Prometheus metrics and pprof server listening on http://127.0.0.1:9758
🥞 Using LevelDB datastore
💭 Using in-memory peerstore
🐲 Spawning 1 heads: 
.2022/07/06 18:17:32 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
🥞 Using default providerstore

🆔 12D3KooWF7oT2Ua29XFw1koqoNmuHUsHZP8U8SMfTTDKw4Kc2NDS
🐝 Swarm listening on /ip4/X.X.X.X/tcp/45223
🐝 Swarm listening on /ip4/127.0.0.1/tcp/45223
🐝 Swarm listening on /ip4/X.X.X.X/udp/58879/quic
🐝 Swarm listening on /ip4/127.0.0.1/udp/58879/quic
🧩 HTTP API listening on http://127.0.0.1:7779
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x13ffa7e]

goroutine 44 [running]:
github.com/libp2p/hydra-booster/providers.(*asyncProvidersFinder).handleRequest(0xc0002943f0, {0x1d829f0, 0xc000d4d600}, {{0x1d82a98, 0xc003807770}, {0x0, 0x0}, {0xc0041055c0, 0x22, 0x30}, ...})

I experienced the same problem on Ubuntu 22.4 and golang to 1.17 (there's a different error with higher version of golang). The previous releases (0.7.3 and 0.7.4) don't seem to work either:

```
panic: qtls.ClientHelloInfo doesn't match

goroutine 1 [running]:
github.com/marten-seemann/qtls-go1-15.init.0()
	~/go/pkg/mod/github.com/marten-seemann/qtls-go1-15@v0.1.0/unsafe.go:20 +0x132
exit status 2

```

--- a/head/head.go
+++ b/head/head.go
@@ -141,7 +141,8 @@ func NewHead(ctx context.Context, options ...opts.Option) (*Head, chan Bootstrap
 
 	var cachingProviderStore *hproviders.CachingProviderStore
 	if cfg.ProvidersFinder != nil && cfg.ReframeAddr == "" {
-		providerStore = hproviders.NewCachingProviderStore(providerStore, providerStore, cfg.ProvidersFinder, nil)
+		cachingProviderStore = hproviders.NewCachingProviderStore(providerStore, providerStore, cfg.ProvidersFinder, nil)
+		providerStore = cachingProviderStore
 	}
 	if cfg.ProvidersFinder != nil && cfg.ReframeAddr != "" {
 		reframeProviderStore, err := hproviders.NewReframeProviderStore(cfg.DelegateHTTPClient, cfg.ReframeAddr)

